documentation/course/e2/README.md

```diff
@@ -1798,6 +1932,21 @@ TODO @tchojnacki: Dodać diagram pakietów, opis architektury.
 
 TODO @jakubzehner: Dodać diagram pakietów, opis architektury i endpointy.
 
+### API
+
+| **Rola**    | **Metoda** | **Endpoint**                   | **Wymagania** | **Opis**                                                                                                                |
+| ----------- | ---------- | ------------------------------ | ------------- | ----------------------------------------------------------------------------------------------------------------------- |
+| `guest`     | `GET`      | `/ext/v1/offers`               | TODO          | Pobranie listy dostępnych ofert biletowych.                                                                             |
+| `guest`     | `GET`      | `/ext/v1/offers/:id`           | TODO          | Pobranie informacji o ofercie biletowej.                                                                                |
+| `passenger` | `GET`      | `/ext/v1/tickets`              | TODO          | Pobranie listy zakupionych biletów.                                                                                     |
+| `passenger` | `POST`     | `/ext/v1/tickets`              | TODO          | Zakup biletu.                                                                                                           |
+| `passenger` | `GET`      | `/ext/v1/tickets/:id`          | TODO          | Pobranie informacji o bilecie.                                                                                          |
+| `passenger` | `POST`     | `/ext/v1/tickets/:id/validate` | TODO          | Walidacja biletu.                                                                                                       |
+| `inspector` | `POST`     | `/ext/v1/tickets/:id/inspect`  | TODO          | Inspekcja biletu.                                                                                                       |
+| `admin`     | `POST`     | `/ext/v1/offers`               | TODO          | Utworzenie nowej oferty biletowej.                                                                                      |
+| `admin`     | `PUT`      | `/ext/v1/offers/:id`           | TODO          | Aktualizacja oferty biletowej, jeśli nie została zakupiona przez żadnego pasażera.                                      |
```

Comment:
```suggestion
```
I'd personally avoid it and just have them delete and create a new one. Seems like really rare conditions for making a dedicated endpoint. 

`TIC/15` seems rather unfortunate.

Alternatively, and I'm making stuff up to make the functional requirement work whereas I'd probably remove it altogether, what if instead of `PUT` it was a `PATCH` endpoint and it:
- created a new TicketOffer with fields specified in request body set to the value from request body, and the other fields set to values from TicketOffer `:id` (something like [nondestructive mutation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record#nondestructive-mutation))
- marked TicketOffer `:id` as inactive?